client/ayon_houdini/plugins/publish/collect_componentbuilder_lop.py

```diff
@@ -0,0 +1,84 @@
+import os
+from typing import List, Tuple
+from pathlib import Path
+
+import pyblish.api
+
+from ayon_core.pipeline import AYONPyblishPluginMixin, PublishError
+from ayon_houdini.api import plugin
+
+import hou
+from pxr import Sdf, UsdUtils
+
+
+def compute_all_dependencies(
+        filepath: str) -> Tuple[list[Sdf.Layer], list[str], list[str]]:
+    """Compute all dependencies for the given USD file."""
+    # Only separated here for better type hints on returned values
+    return UsdUtils.ComputeAllDependencies(filepath)
+
+
+class CollectComponentBuilderLOPs(plugin.HoudiniInstancePlugin,
+                                  AYONPyblishPluginMixin):
+
+    # Run after `CollectResourcesPath`
+    order = pyblish.api.CollectorOrder + 0.496
+    families = ["componentbuilder"]
+    label = "Collect Componentbuilder LOPs"
+
+    def process(self, instance):
+
+        node = hou.node(instance.data["instance_node"])
+
+        # Render the component builder LOPs
+        # TODO: Do we want this? or use existing frames? Usually a Collector
+        #  should not 'extract' but in this case we need the resulting USD
+        #  file.
+        node.cook(force=True)  # required to clear existing errors
+        node.parm("execute").pressButton()
```

Comment:
So this makes sense, however... At the same time I wonder why this isn't triggering on the regular 'render'. There must be a reason that SideFX doesn't do this *always* but has kept it separate. I'm worried that we are too confidently enforcing thumbnail updates where as user may have not wanted this to happen.

Also, wasn't there a toggle on the component builder to make it itself auto generate/save the thumbnail on render without having to trigger it separately?

I feel this is also one of those things that we are deciding for the user more than one may want. However at the same time I personally can't think of a reason if why you wouldn't want the thumbnail updated too (unless the generation of it can be relatively slow or the quality of the default generation sucks?). Anyway, this might be best checked with the client. I am fine with committing this if there is reason to do so, but also would be fine with ignoring it.

Reply:
For reference, this fixes the bug mentioned in https://github.com/ynput/ayon-houdini/pull/163#pullrequestreview-2427741417 where the thumbnail is always expected.
Let me try it once again.

Reply:
> Also, wasn't there a toggle on the component builder to make it itself auto generate/save the thumbnail on render without having to trigger it separately?

Yes, found it. let me quote from [Component Builder Docs](https://www.sidefx.com/docs/houdini/solaris/component_builder.html)
```
If you set Mode to Render or Houdini GL , you can turn on Auto-generate on export to always regenerate the thumbnail when you write out the USD files.
```

But it's not enabled by default.

Reply:
I'm not sure how to tackle this. 
But, I have two ideas: 
1. Exclude the thumbnail from the publish if it doesn't exist. (although the existent thumbnail can be wrong)
2. Force generating the thumbnail, this 👇  might be a better code snippet, but this may be overwhelming as the thumbnail can be heavy as the node supports using Karma.
    ```suggestion
            node.parm("execute").pressButton()
            autothumbnail_parm = node.parm("autothumbnail")
            if not(autothumbnail_parm.isVisible() and autothumbnail_parm.eval()): 
                node.parm("executeviewport").pressButton()
    ```

Reply:
Thanks. I still want to avoid forcing the thumbnail. It's absolutely fine for it to not exist.

In my test runs it ran absolutely fine without thumbnail. It would log a warning but if I recall correctly it published fine. If for whatever reason it's failing to publish then we should fix that. I thought the path would get skipped if it didn't exist, I recall writing that code and logging warnings for that.

Reply:
omg 😅 
I can't reproduce it.